oak_functions_service/src/wasm/mod.rs

```diff
@@ -454,10 +454,25 @@ impl WasmHandler {
             observer,
         })
     }
+}
+
+impl Handler for WasmHandler {
+    type HandlerType = WasmHandler;
```

Comment:
Not sure why this is needed, but we can try and remove it separately anyways.

Reply:
The `Handler` trait has `new_handler(...) -> ?` factory method to create a new handler.
The return type can't be `Self` as `Self` has no known size.
It can't be `Handler`.
Using `dyn Handler` compiles in the trait, but I can't figure out anything sensible to return in the actual implementation.

Another option would be to put things into a `Box`, I guess. But I'm not sure that's any better.

Reply:
Yes I would return a box Dyn, which also goes with my other comment. But up to you and we can do it separately anyway